core/remote/index.js

```diff
@@ -506,8 +506,18 @@ function newDocumentAttributes(
 function mostRecentUpdatedAt /*::<T: Metadata|SavedMetadata> */(
   doc /*: T */
 ) /*: string */ {
-  if (doc.remote && doc.remote.updated_at) {
-    return timestamp.maxDate(doc.updated_at, doc.remote.updated_at)
+  const remoteCreationDate = doc.remote && doc.remote.created_at
+  const remoteModificationDate = doc.remote && doc.remote.updated_at
+  const remoteDate =
+    remoteCreationDate && remoteModificationDate
+      ? timestamp.maxDate(remoteCreationDate, remoteModificationDate)
+      : remoteModificationDate
+      ? remoteModificationDate
+      : remoteCreationDate
+      ? remoteCreationDate
+      : null
```

Comment:
I find this way to structure code hard to read. I would have made something like:

```js
let date = doc.updated_at

const remoteCreationDate = doc.remote && doc.remote.created_at
if (remoteCreationDate) {
  date = timestamp.maxDate(date, remoteCreationDate)
}

const remoteModificationDate = doc.remote && doc.remote.updated_at
if (remoteModificationDate) {
  date = timestamp.maxDate(date, remoteModificationDate)
}

return date
```

But you version works too. It is just a question of style.

Reply:
Your version is probably more readable indeed. I think I'll rewrite this part.